msu/utils/array.nut

```diff
@@ -104,4 +104,15 @@
 			}
 		}
 	}
+
+	// Returns a new array
+	function uniques( _array )
+	{
+		local t = {};
+		foreach (a in _array)
+		{
+			t[a] <- true;
```

Comment:
This will fail if an element in the array is `null`. So we need to add handling for that. I'll work on that next.

Reply:
I have added handling for `null`. However it does not preserve the order in which `null` existed in the original array.

Reply:
I've pushed a new commit based on the algorithm we decided on that preserves order and handles nulls.